src/NavierStokes.hpp

```diff
@@ -0,0 +1,355 @@
+      if (/*p[1] < 0.24*/ 1)
+        values[0] = 3.0;
+      
+      //values[0] = 10.0;
+    }
+
+    virtual double
+    value(const Point<dim> &/*p*/, const unsigned int component = 0) const override
+    {
+      if (/*component == 1 ||*/ component == 0)
+      {
+        if (/*p[1] < 0.24*/1)
+          return 3.0;
+        else
+          return 0.0;
+      }
+      else
+        return 0.0;
+    }
+
+    double
+    get_mean_vel()
+    {
+      return 2.0 /** U_m*/ / 3.0;
+    }
+
+    #else
+
+    virtual void
+    vector_value(const Point<dim> &/*p*/, Vector<double> &values) const override;
+
+    virtual double
+    value(const Point<dim> &/*p*/, const unsigned int component = 0) const override;
+
+    double
+    get_mean_vel();
+
+    #endif
+  };
+
+  class FunctionH : public Function<dim>
+  {
+  public:
+    // Constructor.
+    FunctionH()
+    {
+    }
+
+    virtual double
+    value(const Point<dim> & /*p*/, const unsigned int /*component*/) const override
+    {
+      return 0.0;
+    }
+  };
+
+  class InitialConditions : public Function<dim>
+  {
+  public:
+    InitialConditions()
+      : Function<dim>(dim + 1)
+    {}
+
+    virtual void
+    vector_value(const Point<dim> &/*p*/, Vector<double> &values) const override
+    {
+      //values[0] = 1.0;
+      for (unsigned int i = 0; i < dim + 1; ++i)
+        values[i] = 0.0;
+    }
+
+    virtual double
+    value(const Point<dim> &/*p*/, const unsigned int component = 0) const override
+    {
+      if (component == dim)
+        return 0.0;
+      else
+        return 0.0;
+    }
+  };
+
+  class PreconditionIdentity;
+
+  class PreconditionASIMPLE;
+  
+  class PreconditionAYosida;
+
+
+
+  NavierStokes(const std::string &mesh_file_name_,
+    const unsigned int &degree_velocity_,
+    const unsigned int &degree_pressure_,
+    const double &deltat_,
+    const double &T_,
+    const unsigned int &step_)
+    : mesh_file_name(mesh_file_name_)
+    , mpi_size(Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD))
+    , mpi_rank(Utilities::MPI::this_mpi_process(MPI_COMM_WORLD))
+    , pcout(std::cout, mpi_rank == 0)
+    , degree_velocity(degree_velocity_)
+    , degree_pressure(degree_pressure_)
+    , mesh(MPI_COMM_WORLD)
+    , deltat(deltat_)
+    , T(T_)
+    , step(step_)
+  {}
+
+  void setup();
+
+  void set_re_number(int Re);
+
+  void assemble_matrices();
+
+  void assemble_rhs(const double &time);
+
+  void assemble_static_matrices();
+
+  void solve_time_step(std::ostream &oss);
+
+  void assemble(const double &time);
+
+  void output(const unsigned int &time_step) const;
+
+  void solve(unsigned int time_step = 0);
+
+  void export_data(const unsigned int &time_step);
+
+  void compute_ordered_dofs_indices();
+
+  void import_data(const unsigned int &time_step);
+
+  void post_process(const unsigned int &initial_time_step,
+    const unsigned int &final_time_step, const unsigned int &step);
+
+  void compute_forces(const double &time);
```

Comment:
For functions that compute stuff, unless there's strong reason to do otherwise, it is usually better to have the functions return their result, rather than have them store that result in a state variable (better in the sense of the code organization; this idea is borrowed from the principles of [functional programming](https://en.wikipedia.org/wiki/Functional_programming)).